facebook/delphiFacebook/R/contingency_calculate.R

```diff
@@ -0,0 +1,80 @@
+## Functions used to calculate column aggregations. Each function is meant to be
+## used for a specific response type (binary, numeric, multiselect, multiple choice).
+
+#' Wrapper for `compute_count_response` that adds sample_size. Val is the mean.
+#'
+#' @param response a vector of percentages (100 * cnt / total)
+#' @param weight a vector of sample weights for inverse probability weighting;
+#'   invariant up to a scaling factor
+#' @param sample_size The sample size to use, which may be a non-integer (as
+#'   responses from ZIPs that span geographical boundaries are weighted
+#'   proportionately, and survey weights may also be applied)
+#' @param total_represented Number of people represented in sample, which may
+#'   be a non-integer
+#' 
+#' @return a list of named means and other descriptive statistics
+#'
+#' @export
+compute_numeric <- function(response, weight, sample_size, total_represented)
+{
+  response_mean <- compute_count_response(response, weight, sample_size)
+  response_mean$sample_size <- sample_size
+  response_mean$represented <- total_represented
+  
+  return(response_mean)
+}
+
+#' Wrapper for `compute_binary_response` that adds sample_size. Val is the 
+#' percent `TRUE`.
+#'
+#' @param response a vector of binary (0 or 1) responses
+#' @param weight a vector of sample weights for inverse probability weighting;
+#'   invariant up to a scaling factor
+#' @param sample_size The sample size to use, which may be a non-integer (as
+#'   responses from ZIPs that span geographical boundaries are weighted
+#'   proportionately, and survey weights may also be applied)
+#' @param total_represented Number of people represented in sample, which may
+#'   be a non-integer
+#'      
+#' @return a list of named percentages and other descriptive statistics
+#'
+#' @export
+compute_binary_and_multiselect <- function(response, weight, sample_size, total_represented)
```

Comment:
Multi-select items (where a respondent can select more than one response, e.g. comorbidities) are converted into a series of binary columns, one for each response code. So the calculation for a multi-select item is the same as the binary calculation but applied to multiple columns.

Since these `compute_*` functions are used in constructing the aggregations table, I wanted them to have names that would make it clear when to use them. I could make separate `binary` and `multiselect` versions if you think that would be clearer.